File: html/old/R2/SACO R2 2008 (Solutions)/palindrome.py
#/usr/bin/env python
# SACO 2nd Round 2008
# Palindrome - Keegan and Ben

num = raw_input("Enter number: ")
i = 0
reverse = lambda x: "".join(reversed(x))
 
while num != reverse(num) and i <= 100:
    num = str( int(num) + int(reverse(num)) )
    i+=1

if i > 100:
    print "DOES NOT TERMINATE"
else:
    print i, num

File: html/old/R2/SACO R2 2008 (Solutions)/palindrome.cpp
// SACO Second Round 2008
// Problem: Palindrome
// Author:  Julian Kenwood
#include <iostream>
using namespace std;
int main() {
    int N;
    cout << "Enter N: ";
    cin >> N;
    for(int i = 0; i <= 100; ++i) {
        int rev = 0;
        int temp = N;
        while(temp != 0) {
            rev = rev*10 + temp%10;
            temp /= 10;
        }
        if(N == rev) {
            cout << i << " " << N << endl;
            return 0;
        }
        N+=rev;
   }
   cout << "DOES NOT TERMINATE" << endl;	
   return 0;
}
